pkg/discovery/discoverer.go

```diff
@@ -27,3 +29,11 @@ type FixedDiscoverer []string
 
 // Instances implements Discoverer.
 func (d FixedDiscoverer) Instances() ([]string, error) { return d, nil }
+
+// ErrorDiscoverer yields a discoverer that returns error
+type ErrorDiscoverer []string
```

Comment:
a) I don't think this is a very useful type for public package. If you only use it in tests then define it there.
b) I think you want it to be `type ErrorDiscoverer  error` so that it can `return nil, d` instead of hardcoding error msg